File: Semester3/Advanced Programming Methods/Labs/lab3/src/domain/entities/Student.java
package domain.entities;

public class Student implements HasID<Integer> {
    private Integer id;
    private String name;
    private Integer group;
    private String email;

    /**
     * Student constructor
     * @param id: Integer - id of the student
     * @param name: String - Student's name
     * @param group: Integer - Student's group
     * @param email: String - Student's email
     */
    public Student(Integer id, String name, Integer group, String email){
        this.id = id;
        this.name = name;
        this.group = group;
        this.email = email;
    }


    /***
     * Function to get the name
     * @return - name of the student
     */
    public String getName(){
        return this.name;
    }

    /***
     * Function to get the grouo
     * @return - group of the student
     */
    public Integer getGroup(){
        return this.group;
    }


    /***
     * Function to get the emial of the student
     * @return - email of the student
     */
    public String getEmail(){
        return this.email;
    }


    /**
     * Overriding methods from HasID Interface
     */
    @Override
    public Integer getID() {
        return id;
    }

    @Override
    public void setID(Integer newId) {
        this.id = newId;
    }

    /**
     * Function to set the name
     * @param newName - new name of the student
     */
    public void setName(String newName){
        this.name = newName;
    }


    /**
     * Function to set student's group
     * @param newGroup - new group for the student
     */
    public void setGroup(Integer newGroup){
        this.group = newGroup;
    }


    /**
     * Function to set the email of the student
     * @param newEmail - new email for student
     */
    public void setEmail(String newEmail){
        this.email = newEmail;
    }


    /**
     * Overriding equals method
     */
    @Override
    public boolean equals(Object ot){
        if(ot == this){
            return true;
        }
        if(ot == null){
            return false;
        }
        if(!(ot instanceof Student)){
            return false;
        }

        Student st = (Student)ot;
        return this.id.equals(st.getID());
    }

    /**
     * Overriding toString method
     */
    @Override
    public String toString(){
        return "" + this.id + " " + this.getName() + " " + this.getGroup() + " " + this.getEmail();
    }
}

File: Semester3/Advanced Programming Methods/Labs/StudentsManagementAppC#/StudentsManagementApp/domain/entities/Student.cs
using System;

namespace StudentsManagementApp.domain.entities
{
	public class Student : IHasId<int>
	{

		/// <summary>
		/// Student object constructor
		/// </summary>
		/// <param name="id">Student's id</param>
		/// <param name="name">name of the student</param>
		/// <param name="group">group of the student</param>
		/// <param name="email">email of the student</param>
		public Student(int id, string name, int group, string email)
		{
			Id = id;
			Name = name;
			Group = group;
			Email = email;
		}
		
		/// <summary>
		/// Implementation of IHasId
		/// </summary>
		public int Id { get; set; }
		
		/// <summary>
		/// Name property
		/// </summary>
		public string Name { get; set; }
		
		/// <summary>
		/// Group property
		/// </summary>
		public int Group { get; set;}
		
		/// <summary>
		/// Email property
		/// </summary>
		public string Email { get; set; }

		/// <summary>
		/// Overriding Equals method
		/// </summary>
		/// <param name="o"></param>
		/// <returns>true if the objects are equal, false otherwise</returns>
		public override bool Equals(object o)
		{
			if (!(o is Student)) return false;
			var st = (Student) o;
			return st.Id == Id;

		}


		/// <summary>
		/// Overriding ToString method
		/// </summary>
		/// <returns></returns>
		public override string ToString()
		{
			return "Id: " + Id.ToString() + " Name: " + Name + " Group: " + Group.ToString() + " Email: " + Email;
		}

		/// <summary>
		/// Overriding the GetHashCode method
		/// </summary>
		/// <returns></returns>
		public override int GetHashCode()
		{
			return Id;
		}
	}
	
}